pkg/envoy/ads/stream.go

```diff
@@ -185,41 +185,67 @@ func (s *Server) StreamAggregatedResources(server xds_discovery.AggregatedDiscov
 				xdsUpdatePaths = mapset.NewSetWith(typeURL)
 			}
 
-			err = s.sendResponse(xdsUpdatePaths, proxy, &server, &discoveryRequest, s.cfg)
-			if err != nil {
-				log.Error().Err(err).Msgf("Failed to create and send %s update to Envoy with xDS Certificate SerialNumber=%s on Pod with UID=%s",
-					envoy.XDSShortURINames[typeURL], proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
-				continue
+			// Queue a response job for the request
+			job := proxyResponseJob{
+				typeurls:  xdsUpdatePaths,
+				proxy:     proxy,
+				cfg:       &s.cfg,
+				adsStream: &server,
+				request:   &discoveryRequest,
+				xdsServer: s,
+				done:      make(chan struct{}),
 			}
+			s.workers.AddJob(&job)
```

Comment:
If we change the signature of `AddJob`
from
```go
func (wp *WorkerPool) AddJob(jobs Job) {
```

to
```go
func (wp *WorkerPool) AddJob(jobs Job) <-chan struct{} {
```
We could simplify how this is used.
We AddJob could look like this:
```go
// AddJob posts the job on a worker queue
// Uses Hash underneath to choose worker to post the job to
func (wp *WorkerPool) AddJob(jobs Job) <-chan struct{} {
	wp.workerContext[jobs.Hash()%wp.nWorkers].jobs <- jobs
	return jobs.Done()
}
```

Also I noticed that `AddJob` takes `jobs`, which is plural. It should be singular, since this is not a slice.

Once we do all this we can have:
```go
<-s.workers.AddJob(&job)
```

Reply:
Not bad, not bad at all. I'll do it on a separate commit though (or feel free to augment the interface already), don't want to touch the workerpool interface in this same commit.